frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/GitMacheteGraphTableManager.java

```diff
@@ -224,21 +228,16 @@ private void updateRepository(Path mainDirectoryPath, Path gitDirectoryPath, boo
     }
   }
 
-  private IBranchLayout updateBranchLayoutAndMacheteFilePath() throws MacheteFileParseException {
-    GitRepository gitRepository = gitRepositorySelectionProvider.getSelectedRepository();
-    Path macheteFilePath = getMacheteFilePath(gitRepository);
-    IBranchLayout branchLayout = createBranchLayout(macheteFilePath);
-    graphTable.setBranchLayout(branchLayout);
-    graphTable.setMacheteFilePath(macheteFilePath);
-    return branchLayout;
-  }
-
-  private IBranchLayout createBranchLayout(Path branchLayoutFilePath) throws MacheteFileParseException {
-    return Try.of(() -> branchLayoutParserFactory.create(branchLayoutFilePath).parse())
+  private IBranchLayout createBranchLayout(IBranchLayoutManager branchLayoutManager) throws MacheteFileReaderException {
+    IBranchLayout branchLayout = Try.of(() -> branchLayoutManager.getReader().read())
         .getOrElseThrow(e -> {
           Option<@Positive Integer> errorLine = ((BranchLayoutException) e).getErrorLine();
-          return new MacheteFileParseException("Error occurred while parsing machete file ${branchLayoutFilePath}" +
+          return new MacheteFileReaderException("Error occurred while parsing machete file" +
               (errorLine.isDefined() ? " in line ${errorLine.get()}" : ""), e);
         });
+
+    graphTable.setBranchLayout(branchLayout);
```

Comment:
I'd keep this method non-side-effecting, call `setBranchLayout` up around line 210